### PR TITLE
Update the minimum version of CMake and tidy up

### PR DESCRIPTION
**Pull request by riemass**
Tidied up some old CMake rules. 
Not so sure about setting the CMake policy max to 3.27 (current release) but it seems to compile without problems.

### CODE REVIEW

**Comment by riemass** [comment]
The windows action started failing with `python3.exe: No module named robot` after the changes, which makes little sense to me. 
I've run the pipeline on another PR to check if it's related to the changes.

File: robot/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+)
+
+if (NOT ${EXIT_CODE} EQUAL 0) 
+  message(
+    FATAL_ERROR
+    "The python package 'robotframework' could not be found"
+  )
+endif()
+
+
```
Comment (Neverlord) [inline robot/CMakeLists.txt:16]:
```suggestion

```

File: robot/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+execute_process(
+  COMMAND ${Python_EXECUTABLE} -c "import robot"
+  RESULT_VARIABLE EXIT_CODE
+  OUTPUT_QUIET
+)
+
+if (NOT ${EXIT_CODE} EQUAL 0) 
+  message(
+    FATAL_ERROR
+    "The python package 'robotframework' could not be found"
```
Comment (Neverlord) [inline robot/CMakeLists.txt:12]: Please use the active voice:

```suggestion
    "Failed to import the 'robot' Python package"
```

**Review by Neverlord — CHANGES_REQUESTED** [review]
Just some nitpicks.

**Review by Neverlord — APPROVED** [review]

**Pull request merged**